packages/flutter_fuchsia/lib/src/dart/fuchsia_dart_vm.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:io';
+
+import 'package:json_rpc_2/json_rpc_2.dart' as json_rpc;
+import 'package:meta/meta.dart';
+import 'package:logging/logging.dart';
+import 'package:web_socket_channel/io.dart';
+
+const Duration _kConnectTimeout = const Duration(seconds: 30);
+
+const Duration _kReconnectAttemptInterval = const Duration(seconds: 3);
+
+const Duration _kRpcTimeout = const Duration(seconds: 5);
+
+final Logger _log = new Logger('FuchsiaDartVm');
+
+/// Simple JSON RPC peer with a Fuchsia Dart VM service instance. Wraps most
+/// methods around a JSON RPC peer class.
+class FuchsiaDartVm {
+  final json_rpc.Peer _peer;
+
+  FuchsiaDartVm._(this._peer);
+
+  /// Attempts to connect to the given [Uri]. Throws an error if unable to
+  /// connect.
+  static Future<FuchsiaDartVm> connect(Uri uri) async {
+    final Stopwatch timer = new Stopwatch()..start();
+    if (uri.scheme == 'http') uri = uri.replace(scheme: 'ws', path: '/ws');
+    return await _attemptConnection(uri, timer);
+  }
+
+  static Future<FuchsiaDartVm> _attemptConnection(
+      final Uri uri, final Stopwatch runningTimer) async {
+    WebSocket socket;
+    try {
+      socket = await WebSocket.connect(uri.toString());
+      json_rpc.Peer peer =
+          new json_rpc.Peer(new IOWebSocketChannel(socket).cast())..listen();
+      return new FuchsiaDartVm._(peer);
+    } catch (e) {
+      await socket?.close();
+      if (runningTimer.elapse < _kConnectTimeout) {
+        log.info('Attempting to reconnect');
+        await new Future<Null>.delayed(_kReconnectAttemptInterval);
+        return FuchsiaDartVm._attemptConnection(uri, runningTimer);
+      } else {
+        log.critical('Connection to Fuchsia\'s Dart VM timed out.');
+        rethrow;
+      }
+    }
+  }
+
+  /// Invokes a raw RPC command with the VM service.
+  Future<Map<String, dynamic>> invokeRpc(String function,
+      {Map<String, dynamic> params, Duration timeout}) async {
+    return _peer.sendRequest(function, params ?? {});
+  }
+
+  /// Returns a list of flutter views by name. If the name of a flutter view
+  /// cannot be determined (there is no associated isolate), then the flutter
+  /// view's ID will be added instead.
```

Comment:
Somewhere in the API docs we should clearly define what is a "flutter view", "ID" and other Fuchsia-specific concepts. It's ok if there's no clear (or stable) definition yet, but anything that clarifies it for the user should be useful.

Reply:
To be totally frank, I don't think I know enough about the specifics of those things yet to clearly define them (mainly the ID.  It is something I'm currently using because it is available and a string).